library/data-model/src/data_storage/storageFunctions.ts

```diff
@@ -218,12 +220,12 @@ export async function listFAIMSRecordRevisions({
 export async function listFAIMSProjectRevisions({
   dataDb,
 }: {
-  dataDb: DataDbType;
+  dataDb: DataDbType | RevisionDbType;
 }): Promise<ProjectRevisionListing> {
   try {
     // get all revision
     const result = await queryCouch<Revision>({
-      db: dataDb,
+      db: dataDb as RevisionDbType,
```

Comment:
all this type casting is scaring me a bit - why do we need to do this now, and not previously?

Reply:
I'm not sure why it worked with the native type declarations but not these, my types are compatible but I couldn't get a build to work without this.